contentcuration/contentcuration/utils/garbage_collect.py

```diff
@@ -42,6 +48,48 @@ def get_deleted_chefs_root():
     return deleted_chefs_node
 
 
+def _clean_up_files(contentnode_ids):
+    """
+    Clean up the files (both in the DB and in object storage)
+    associated with the `contentnode_ids` iterable that are
+    not pointed by any other contentnode.
+    """
+    files = File.objects.filter(contentnode__in=contentnode_ids)
+    files_on_storage = files.values_list("file_on_disk", flat=True)
+
+    for disk_file_path in files_on_storage:
```

Comment:
Reinstating the file deletion behaviour here does make sense in the context of the PR - but would be good to have some tests on this function just to guarantee that we are definitely _not_ deleting files that are being used in other channels - regardless of how simple the code may seem here.

Reply:
The reinstation of file deletion plays a critical role in garbage collection so I was determined to make sure it doesn't bring in bad things. 

Thankfully, there were tests already written for the case when two contentnodes share the same file and there are more file deletion test cases under `CleanUpContentNodesTestCase` that cover other behaviours also:

https://github.com/learningequality/studio/blob/172eb66551a9bab638f2759a6eceb61c654a0410/contentcuration/contentcuration/tests/utils/test_garbage_collect.py#L195 
 
https://github.com/learningequality/studio/blob/172eb66551a9bab638f2759a6eceb61c654a0410/contentcuration/contentcuration/tests/utils/test_garbage_collect.py#L237-L241

I have made sure that changing the file deletion behaviour fails the above tests. Since the current code passes the above tests, I am confident with my changes.

@rtibbles sir, if you think there's a case that's remaining to be tested...? then I'll definitely love to write test for that case.

Reply:
@rtibbles umm, sir, thoughts?